### PR TITLE
feat(type): add isSet

### PR DESCRIPTION
**Pull request by realSaddy**
#1110

### CODE REVIEW

File: src/isSet.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Predicate for determining if a provided value is an instance of a Set.
+ *
+ * @func isSet
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {boolean}
+ * @see {@link RA.isSet}
```
Comment (char0n) [inline src/isSet.js]: `@see {@link RA.isMap|isMap}` - it doesn't make sense to link to itself. Somebody else is working on `isMap` PR, so let's link to it.

File: types/index.d.ts [hunk 2]
```diff
@@ -362,6 +362,11 @@ declare namespace RamdaAdjunct {
          */
         isRegExp(val: any): boolean;
 
+        /**
+         * Checks if value is `Set`.
+         */
+        isSet(val: any): boolean;
```
Comment (char0n) [inline types/index.d.ts]:
```ts
isSet(val: any): val is Set

File: types/test/isSet.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+RA.isSet(NaN); // $ExpectType boolean
+RA.isSet({}); // $ExpectType boolean
+RA.isSet([]); // $ExpectType boolean
+RA.isSet(() => {}); // $ExpectType boolean
+RA.isSet('string'); // $ExpectType boolean
+RA.isSet(1); // $ExpectType boolean
+RA.isSet(Object()); // $ExpectType boolean
+RA.isSet(Object(5)); // $ExpectType boolean
+RA.isSet(new Map()); // $ExpectType boolean
+RA.isSet(new Set()); // $ExpectType boolean
```
Comment (char0n) [inline types/test/isSet.ts]: Move doing assertions about set on top. We always want to have positive stuff on top, and then continue with all the negative ;]

File: test/isSet.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import * as RA from '../src';
+import eq from './shared/eq';
+import Symbol from './shared/Symbol';
+import args from './shared/arguments';
+
+describe('isSet', function() {
```
Comment (char0n) [inline test/isSet.js]: Can you pls aligned your test with the new conventions we have for writing tests demonstrated here? https://github.com/char0n/ramda-adjunct/blob/9cd86e7d06de667c14f4fdc8661d0fd4a4c99dc3/test/isNotBoolean.js#L5

We also missing the test for currying completely.

File: src/isSet.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * @example
+ *
+ * RA.isSet(new Map()); //=> false
+ * RA.isSet(new Set()); //=> true
+ * RA.isSet(new Set([1,2]); //=> true
+ * RA.isSet(new Object()); //=> false
+ */
+
+const isSet = curry1(val => is(Set, val));
```
Comment (char0n) [inline src/isSet.js]: We're not using `R.is` in this library anymore. It's has a lot of problems. One of the main problem is described here: https://github.com/char0n/ramda-adjunct/issues/724

instead we prefer to use [tagChecking](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag) which solves the problem. 

Pls aligned your implementation with `isDate` function - https://github.com/char0n/ramda-adjunct/blob/96d0772005ddf262ef2871ca5484dc156eb49805/src/isDate.js#L21

File: src/isSet.js (hunk 1 above)
Comment (realSaddy) [reply src/isSet.js]: I think I have done so now

File: types/index.d.ts (hunk 2 above)
Comment (realSaddy) [reply types/index.d.ts]: I think I have done so now. I had to do `val is Set<any>` due to it requiring a type

File: types/test/isSet.ts (hunk 3 above)
Comment (realSaddy) [reply types/test/isSet.ts]: I think I have done so now

File: test/isSet.js (hunk 4 above)
Comment (realSaddy) [reply test/isSet.js]: Moved it to the new standard. I am confused about testing for currying. I tried to check other examples where it was used, for example isNotBoolean, however, their tests don't have anything that jumps out as different from what I wrote. Could you elaborate if I got it incorrect? Thank you

File: src/isSet.js (hunk 5 above)
Comment (realSaddy) [reply src/isSet.js]: Moved over to the pipe, type, identical standard

**Comment by realSaddy** [comment]
Made changes to fix these issues; thank you for hosting this opportunity! If anything more needs to be changed, please tell me, and I'll get on it

File: types/index.d.ts (hunk 2 above)
Comment (char0n) [reply types/index.d.ts]: Yeah, that's good enought.

File: test/isSet.js (hunk 4 above)
Comment (char0n) [reply test/isSet.js]: It's possible that I overlook this test `should support placeholder to specify "gaps"`. Now i can see it, so marking this as resolved.

**Comment by char0n** [comment]
Nice job. Thank you for contributing!

**Pull request merged**